src/ngraph/serializer.cpp

```diff
@@ -3111,9 +3132,15 @@ json JSONSerializer::serialize_node(const Node& n)
     json inputs = json::array();
     json control_deps = json::array();
     json outputs = json::array();
+    json input_attributes = json::array();
+    json output_attributes = json::array();
 
     for (auto& input : n.inputs())
     {
+        json attributes = json::object();
+        attributes["max_partial_shape"] = write_partial_shape(input.get_max_partial_shape());
```

Comment:
Seems to me this really wants to be a partial_shape object with min and max values in it. Same comment for output

Reply:
I think the min/max would be in the `Dimension` object; a partial shape is based on a vector of `Dimension` objects.

Reply:
If the Dimension has a min and max and min<max it's dynamic, if min==max it's static. I am going to try a new PR that uses this approach; I think it will be compatible with existing shape inference if they are using the arithmetic operators on Dimension.